src/high_scores.rs

```diff
@@ -82,10 +150,9 @@ fn upgrade_if_needed(filename: &str) -> Result<(), AnyErrorThreadSafe> {
 
     if let Some(old_version) = first_line.strip_prefix(HEADER_PREFIX) {
         match old_version {
-            "1" | "2" | "3" if VERSION == "4" => {
-                // Previous formats are compatible with v4
+            "1" | "2" | "3" | "4" => {
                 append_update_comment(filename, old_version)?;
-                update_version_number(filename)?;
+                update_from_v4_or_older(filename)?;
```

Comment:
Appending update comment last would make more sense, even though it doesn't really matter as upgrading preserves comments